pandas/core/algorithms.py

```diff
@@ -759,7 +759,7 @@ def factorize(
     else:
         dtype = values.dtype
         values = _ensure_data(values)
-        na_value: Scalar
+        na_value: Scalar | None
```

Comment:
Scalar doesn't include None?

Reply:
No, None is not included (datetime is also not included).